examples/api/lib/material/theme/theme_extension.1..dart

```diff
@@ -0,0 +1,134 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+// Flutter code sample for ThemeExtension
+
+import 'package:flutter/material.dart';
+
+@immutable
+class MyColors implements ThemeExtension {
+  const MyColors({
+    this.blue,
+    this.red,
+  });
+
+  final Color? blue;
+  final Color? red;
+
+  @override
+  MyColors copyWith({Color? red, Color? blue}) {
+    return MyColors(
+      blue: blue ?? this.blue,
+      red: red ?? this.red,
+    );
+  }
+
+  @override
+  MyColors lerp(ThemeExtension? other, double t) {
+    if (other is MyColors) {
```

Comment:
Sometimes it's easier to read if the short-circuit is first
```
if (other is! MyColors) {
  return this;
}

return MyColors...
```

Reply:
nit: maybe even 
if (other is! MyColors) return this;

Reply:
ah neat, didn't know the ! could go after is

Reply:
ah yeah, reason I had it that way is for type-checking to work